Packs/GenericSQL/Integrations/GenericSQL/GenericSQL.py

```diff
@@ -129,12 +133,21 @@ def _create_engine_and_connect(self) -> sqlalchemy.engine.base.Connection:
                      port=arg_to_number(self.port),
                      database=self.dbname,
                      query=self.connect_parameters)
+
         if self.ssl_connect:
             if self.dialect == POSTGRES_SQL:
                 ssl_connection = {'sslmode': 'require'}
             else:
                 ssl_connection = {'ssl': {'ssl-mode': 'preferred'}}  # type: ignore[dict-item]
-        engine: sqlalchemy.engine.Engine = None
+
+        # Teradata has a unique connection, unlike the others with URL object
+        if self.dialect == TERADATA:
+            if self.use_ldap:
+                db_url = f'teradatasql://{self.username}:{self.password}@{self.host}:{self.port}/?logmech=LDAP'
+            else:
+                db_url = f'teradatasql://{self.host}:{self.port}/?user={self.username}&password={self.password}'
+            demisto.debug('Initializing engine using the Teradata dialect')
```

Comment:
let's put this, together with `db_url = URL(...)`, in their own method called `_generate_db_url`

Reply:
Done